src/util/tabTools.ts

```diff
@@ -0,0 +1,22 @@
+// Libs
+import idx from 'idx';
+// Constants
+import { FAIL } from "../constants/transactions";
+
+type W3Error = {
+    message: string,
+    stack: string
+}
+
+export const errorToast = (error: W3Error, value: string, openToast: (identifier: string, status: string, message: string) => void, defaultToast: () => void) => {
+    const message = idx(error, _ => _.message);
+    if (message && message.includes("MetaMask Tx Signature: User denied transaction signature.")) {
```

Comment:
should we add "MetaMask" by name? can we generalize it somehow? because technically our dapp can talk to any browser based client, right?

Reply:
Agreed - I'd love to make it more generic, but W3 is just returning Strings with error messages and I'm not aware of whether MetaMask alternatives provide a message that is similar enough to match this generic case.